python/events/evaluation_tools/events/event_detection/decomposition.py

```diff
@@ -22,11 +25,49 @@
 from typing import Union
 import datetime
 
+def rolling_minimum(
+    series: pd.Series,
+    window: Union[int, pd.tseries.offsets.DateOffset, pd.Index]
+    ) -> pd.Series:
+    """Model the trend in a streamflow time series using a rolling 
+        minimum filter. 
+        Return the trend of a streamflow time series.
+        
+        Parameters
+        ----------
+        series: pandas.Series with a DateTimeIndex, required
+            The original streamflow time series.
+        window: int, offset, or BaseIndexer subclass, required
+            Size of the moving window for `pandas.Series.rolling.min`.
+            This filter is used to model the trend in `series`.
+            
+        Returns
+        -------
+        trend: pandas.Series
+            Approximate baseflow trend of original series.
+        
+        """
```

Comment:
It seems appropriate to check if the index of the series is a `pd.DatetimeIndex`. This was done in `detrend_streamflow`. So something like:


```python
if isinstance(series.index, pd.DatetimeIndex) is False:
  # Throw an exception
```

Reply:
7af6524 addresses this.

Reply:
Is there a conventional or standard way to check for this?

Examples:
```python
if isinstance(series.index, pd.DatetimeIndex) is False:
  # Throw an exception

if not isinstance(series.index, pd.DatetimeIndex):
  # Throw an exception

if type(series.index) != pd.DatetimeIndex:
  # Throw an exception
```